src/uu/expr/src/syntax_tree.rs

```diff
@@ -482,16 +482,15 @@ fn infix_operator_and(values: &[String]) -> String {
 
 fn operator_match(values: &[String]) -> Result<String, String> {
     assert!(values.len() == 2);
-    let re = Regex::with_options(&values[1], RegexOptions::REGEX_OPTION_NONE, Syntax::grep())
-        .map_err(|err| err.description().to_string())?;
-    Ok(if re.captures_len() > 0 {
+    let re = Regex::new(&values[1]).map_err(|err| err.to_string())?;
+    Ok(if re.captures_len() > 1 {
```

Comment:
I guess you changed it from `> 0` to `> 1` because of the `.get(1)` on line 488?

Reply:
I think this is a difference between the regex engines. `regex` counts captures including the entire match, but onig doesn't (at least that's what I think).

Reply:
Makes sense.